command/loadbot/execution.go

```diff
@@ -196,20 +232,65 @@ func (l *Loadbot) Run() error {
 		return fmt.Errorf("failed to extract sender account: %v", err)
 	}
 
-	client, err := createJsonRpcClient(l.cfg.JSONRPC, l.cfg.MaxConns)
+	jsonClient, err := createJsonRpcClient(l.cfg.JSONRPC, l.cfg.MaxConns)
+	if err != nil {
+		return fmt.Errorf("an error has occured while creating JSON-RPC client: %v", err)
+	}
+
+	grpcClient, err := createGRPCClient(l.cfg.GRPC)
 	if err != nil {
 		return fmt.Errorf("an error has occured while creating JSON-RPC client: %v", err)
 	}
 
 	defer func(client *jsonrpc.Client) {
 		_ = shutdownClient(client)
-	}(client)
+	}(jsonClient)
 
-	nonce, err := getInitialSenderNonce(client, sender.Address)
+	nonce, err := getInitialSenderNonce(jsonClient, sender.Address)
 	if err != nil {
-		return fmt.Errorf("an error occured while getting initial sender nonce: %v", err)
+		return fmt.Errorf("unable to get initial sender nonce: %v", err)
 	}
 
+	gasPrice, err := getAverageGasPrice(jsonClient)
+	if err != nil {
+		return fmt.Errorf("unable to get average gas price: %v", err)
+	}
```

Comment:
You're completely right.

We used to hardcode it before, this is just a more dynamic solution.
What do you suggest?

Reply:
I don't think it's a good to fetch the average gas price every time you create new transaction because it affects thorughput.
Firstly, it's better to be able to pass gas price from flags because sometimes we want to give much enough gas price to process transactions as fast as possible.

Reply:
I agree 👍 

I've added the ability for you to specify both the gas limit and gas price if you want, but if you don't specify something the value will be fetched from the network:

a0efd89